src/App/App.js

```diff
@@ -1,44 +1,57 @@
 import React, { Component } from 'react';
 import './App.scss';
 import Header from '../Header/Header';
-import movieData from '../Data/movieData';
-import fakeMovieDetails from '../Data/fakeMovieDetails';
 import MovieDetails from '../MovieDetails/MovieDetails';
 import tomatillo from '../assets/tomatillo.svg';
 import MovieList from '../MovieList/MovieList';
 import MovieCard from '../MovieCard/MovieCard';
+import { fetchAllMovies, fetchSingleMovie, fetchVideo } from '../Data/apiCalls';
 import logo from '../assets/logo.svg';
-import tomatillo from '../assets/tomatillo.svg';
-
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movieData: movieData.movies,
-      movieDetails: fakeMovieDetails.movie,
-      pageLocation: 0
-      // movieData: [],
-      // movieDetails: [],
-      // filteredMovies: [],
-      // error: ''
+      movieData: [], //all of the movie on home page
+      movieDetails: [], //singe movie
+      filteredMovies: [],
+      pageLocation: 0,
+      // currentMovie: {},
+      isLoading: true,
+      error: '',
+
     }
   }
 
+  componentDidMount() {
+    fetchAllMovies()
+      .then(movies => this.setState({ movieData: movies.movies }))
+      .catch(err => this.setState({ error: 'Something went wrong' }))
+  }
+
   // filterMovies function here, also need to pass filterMovies funtion to header
 
   calculatePercent = rating => {
-     return (rating.toFixed(1) * 10);
-   }
+    return (rating.toFixed(1) * 10);
+  }
 
-   clickHandler = (id) => {
-     this.setState({ pageLocation: id})
-   }
+  resetMovie = () => {
+
+  }
+
+  clickHandler = (id) => {
+    this.setState({ movieDetails: [] })
+    this.setState({ pageLocation: id })
+    fetchSingleMovie(id)
+      .then(movie => {
+        this.setState({ movieDetails: movie.movie });
+      })
+  }
 
   render() {
     return (
+      // <>
       <main className='App'>
-
-        <Header click={this.clickHandler}/>
+        <Header click={this.clickHandler} />
```

Comment:
I haven't seen just a click attribute before - why are you using this over onClick with a callback?